test/jasmine/tests/sankey_test.js

```diff
@@ -248,4 +254,61 @@ describe('sankey tests', function() {
             });
         });
     });
+
+    describe('lifecycle methods', function() {
+
+        afterEach(destroyGraphDiv);
+
+        it('Plotly.deleteTraces with two traces removes the deleted plot', function(done) {
+
+            var gd = createGraphDiv();
+            var mockCopy = Lib.extendDeep({}, mock);
+            var mockCopy2 = Lib.extendDeep({}, mockDark);
+
+            Plotly.plot(gd, mockCopy)
+                .then(function() {
+                    expect(gd.data.length).toEqual(1);
+                    expect(d3.selectAll('.sankey').size()).toEqual(1);
+                    return Plotly.plot(gd, mockCopy2);
```

Comment:
testing `addTraces` would be better. Sequential `Plotly.plot` call is a deprecated pattern.